extension/extension.proto

```diff
@@ -0,0 +1,174 @@
+syntax = "proto3";
+
+package gnoi.extension;
+
+import "types/types.proto";
+
+option go_package = "github.com/openconfig/gnoi/extension";
+
+option (types.gnoi_version) = "0.1.0";
+
+// Extension is a service for managing extensions on the system.
+// Only one RPC (SetSignatureVerification, FinalizeExtensions,
+// InstallExtension, UninstallExtension)
+// may be in progress at any given time
+service Extension{
```

Comment:
what is meant by an `extension`? Is it a software bundle that runs on a NOS?
The term seems to be not so well known...

Reply:
an extension here is just an bundle of arbitrary files (e.g. RPMs) which can be installed and applied over the system to modify the behaviour of some services on the system. 
This is very generic, theres nothing specific here about what an extension must do or any requirements for the extension file itself. 
I've added a comment in the next commit https://github.com/openconfig/gnoi/pull/160/commits/ea29e019b9b3d985d5b3472140f60aed1d8c4106

As an example of this: 
Some network operator experiences a bug with BGP services, and a vendor may provide that network operator with an extension which can be installed to apply a fix for that bug. The network operator transfers the extension (verifying that it was signed by the device manufacturer if desired) and then activates it, which applies the fix to BGP.
 
Network operators may prefer doing this rather than performing a full upgrade to a new OS version using `gnoi OS`, as an extension can be applied without installing a new software image, and it doesnt affect any unrelated services

Reply:
@hellt - 

the notion of extension installation is pretty common across NOS.  this might take the form of vendor-provided (and signed) extensions or the installation of common linux utilities that extend the base capabilities of the NOS.

examples include: 
- juniper/HPE junos(-evolved) - [How to Install Third-Party Software on Devices Running Junos OS Evolved | Junos OS Evolved | Juniper Networks](https://www.juniper.net/documentation/us/en/software/junos/junos-install-upgrade-evo/topics/concept/junos-evolved-install-third-party-software-overview.html)
- cisco ios-xr - [Cisco IOS XR Setup and Upgrade Guide for Cisco 8000 Series Routers - Manage the Router \[Cisco 8000 Series Routers\] - Cisco](https://www.cisco.com/c/en/us/td/docs/iosxr/cisco8000/b-setup-and-upgrade-cisco8k/manage-software.html)
- arista [EOS 4.31.1F - Managing EOS Extensions - Arista](https://www.arista.com/en/um-eos/eos-managing-eos-extensions?searchword=eos%20section%206%207%20switch%20administration%20commands)

Reply:
@sulrich yes, the concept is clear, it is just that "extension" seems to be a peculiar choice for a package.

But maybe it is just me :)